room_allocation.py

```diff
@@ -79,21 +89,29 @@ def add_person(self, person_name, typ, wants_accomodation=False):
                 habitable_rooms.append(room)
 
         try:
-            room = random.choice(habitable_rooms)
-            room.people_in_room.append(person)
-            return person
+            if habitable_rooms:
```

Comment:
Because of the if statement, the index error will never occur hence the try except block does not have any functionality. The if ensures that the random.choice function never executes if habitable_rooms is empty